spec/controllers/admin/users_controller_spec.rb

```diff
@@ -52,6 +52,27 @@ def login_user
       expect(assigns(:users)).to match_array([])
       expect(response).to be_successful
     end
+
+    context ".json" do
+      render_views
+
+      it "fetches users for search term" do
+        create(:user, full_name: "Doctor Jack")
+        create(:user, full_name: "Jack")
+        search_query = "Doctor"
+
+        get :index, format: :json, params: {search_query: search_query}
+
+        expect(response).to be_successful
+        expect(JSON(response.body).first["full_name"]).to eq "Doctor Jack"
+      end
+
+      specify do
+        expect(User).to receive(:search_by_name_or_teleconsultation_phone_number).with(search_query)
```

Comment:
Should we `allow` this to receive this and return a specific user? We can then assert that the user that's found is sent in the response correctly.